platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform;
+
+import static com.swirlds.logging.LogMarker.EXCEPTION;
+import static com.swirlds.logging.LogMarker.STARTUP;
+
+import com.swirlds.common.system.SoftwareVersion;
+import com.swirlds.common.system.SwirldState;
+import com.swirlds.common.system.address.AddressBook;
+import com.swirlds.common.system.address.AddressBookValidator;
+import com.swirlds.platform.state.signed.SignedState;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+import java.util.function.Supplier;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Determines the initial address book to use at the start of day and validates it.
+ */
+public class AddressBookInitializer {
```

Comment:
Theoretically, this class could just be a static method, where some params are passed in, and an address book pops out. Just a thought

Reply:
There are two items retrievable from the `AddressBookInitializer`, the initial address book, and the directory for where the address books used in the initialization were recorded.    Having this as an instantiable class allows it to be mocked with alternate behavior.